### ISSUE TITLE
Toast id is not available to custom content 

### ISSUE THREAD

**Issue by benhodgson87** [teodosii/react-notifications-component#35]
As far as I can tell, currently if a custom content component is provided, no props are made available to it. This means that as no close icon is shown it requires _dismiss.click_ to be true to allow the user to dismiss the toast - which clashes if the custom toast contains something like a react-router Link or any other click handler.

Ideally `content` should pass down the id of the toast as a prop to allow the custom component to call the `removeNotification(id)` method on itself.

eg:
```jsx
store.addNotification({
  content: ({ id }) => (
    <div>
      Item added to <Link to="/basket">your basket</Link>
      <br />
      <button onClick={() => store.removeNotification(id)}>× Dismiss</button>
    </div>
  )
})
```

**Comment by teodosii** [comment]
Hmm, you're correct. Will do a fix these days

**Comment by teodosii** [comment]
Fixed, will be released in `2.2.0`

**Comment by jabidof** [comment]
Hi,
Nice lib.

I need the notification ID for different usage.
I tried above example without success :-(

I keep getting below errors.

```
TypeError: Cannot read property 'isReactComponent' of undefined
isClassComponent
src/utils/validators.js:12
   9 | const isArray = object => !isNull(object) && object.constructor === Array;
  10 | 
  11 | function isClassComponent(component) {
> 12 |   return typeof component === 'function' && !!component.prototype.isReactComponent;
```

Seems the integration of React component in "content" prop is the issue.

Could you provide a working example with custom content?

Thanks
